scripts/HC01-JP/c100287004.lua

```diff
@@ -0,0 +1,127 @@
+--伝説の剣闘士 カオス・ソルジャー
+--Script by JSY1728
+local s,id,o=GetID()
+function s.initial_effect(c)
+	c:EnableReviveLimit()
+	--Special Summon Condition
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_SINGLE)
+	e1:SetProperty(EFFECT_FLAG_CANNOT_DISABLE+EFFECT_FLAG_UNCOPYABLE)
+	e1:SetCode(EFFECT_SPSUMMON_CONDITION)
+	e1:SetValue(aux.ritlimit)
+	c:RegisterEffect(e1)
+	--Deck Search
+	local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(id,0))
+	e2:SetCategory(CATEGORY_TOHAND)
+	e2:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_TRIGGER_O)
+	e2:SetCode(EVENT_PREDRAW)
+	e2:SetRange(LOCATION_HAND)
+	e2:SetCondition(s.thcon)
+	e2:SetCost(s.thcost)
+	e2:SetTarget(s.thtg)
+	e2:SetOperation(s.thop)
+	c:RegisterEffect(e2)
+	--Activate Limit
+	local e3=Effect.CreateEffect(c)
+	e3:SetType(EFFECT_TYPE_FIELD)
+	e3:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e3:SetCode(EFFECT_CANNOT_ACTIVATE)
+	e3:SetRange(LOCATION_MZONE)
+	e3:SetTargetRange(0,1)
+	e3:SetCondition(s.actcon)
+	e3:SetValue(1)
+	c:RegisterEffect(e3)
+	--Return to Hand
+	local e4=Effect.CreateEffect(c)
+	e4:SetDescription(aux.Stringid(id,1))
+	e4:SetCategory(CATEGORY_TOHAND)
+	e4:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e4:SetCode(EVENT_BATTLE_DESTROYING)
+	e4:SetCondition(s.rhcon)
+	e4:SetTarget(s.rhtg)
+	e4:SetOperation(s.rhop)
+	c:RegisterEffect(e4)
+end
+function s.thcon(e,tp,eg,ep,ev,re,r,rp)
+	return Duel.GetTurnPlayer()==tp and Duel.GetFieldGroupCount(tp,LOCATION_DECK,0)>0
+		and Duel.GetDrawCount(tp)>0

```

Comment:
This effect always activates in Turn 1, even if the player is going 1st and therefore cannot conduct their normal Draw.
```suggestion
		and Duel.GetDrawCount(tp)>0 and Duel.GetTurnCount()>1
```

Reply:
Sorry, I mistake.
I will fix this.

Reply:
also any other problem?

Reply:
> This effect always activates in Turn 1, even if the player is going 1st beyond Master Rule 3 and therefore cannot conduct their normal Draw.

Sorry, I mistake.
I will fix this.
also any other problem?

Reply:
Why does the card need to remain revealed and in the hand for the effect to resolve?

Reply:
> Why does the card need to remain revealed and in the hand for the effect to resolve?

you mean the first effect? or any other?

Reply:
The 1st effect

Reply:
> The 1st effect

This script is bad? (not good?)

Reply:
I was answering your question; also, I just posted some more changes to your file. Again, why does BLS need to remain revealed and in the hand for the 1st effect to resolve, and why is the normal Draw given up when the effect activates instead of resolves?